etl/steps/data/meadow/usgs/2024-07-15/historical_statistics_for_mineral_and_material_commodities.py

```diff
@@ -0,0 +1,174 @@
+
+    # Remove notes from the table.
+    df = df[df["Year"].astype(str).str.isdigit()].reset_index(drop=True)
+
+    # Rename some columns so they are consistent with other tables.
+    df = df.rename(
+        columns={
+            "Unit value ($/t)": "Unit value $/t",
+            "Unit value (98 $/t)": "Unit value 98$/t",
+            "Unit value (98$/t)": "Unit value 98$/t",
+            # Note that we inform if the weight is gross in the "unit" column.
+            "World production (gross weight)": "World production",
+        },
+        errors="ignore",
+    )
+
+    return df
+
+
+def combine_data_for_all_commodities(supply_demand_data: dict) -> pd.DataFrame:
+    # Initialize a dataframe that will combine the data for all commodities.
+    combined = pd.DataFrame()
+    for commodity, data in supply_demand_data.items():
+        if commodity == "abrasives__natural__discontinued__see_garnet__industrial":
+            # This file has a different format, with two tables in the same sheet.
+            # For now, simply skip it.
+            continue
+
+        if commodity == "nickel":
+            # For commodities with multiple sheets, the sheets correspond to different commodities.
+            # For example, "bauxite_and_alumina" has a sheet for "Bauxite" and another for "Alumina".
+            # However, for "nickel", the sheets show more detailed nickel data, which for now we don't need.
+            sheet_names = ["Nickel"]
+        else:
+            sheet_names = data.sheet_names
+
+        for sheet_name in sheet_names:
+            if sheet_name == "Sheet1":
+                # This spurious empty sheet appears sometimes; simply skip it.
+                continue
+            df = clean_sheet_data(data=data, commodity=commodity, sheet_name=sheet_name)
+
+            # Add the dataframe for the current commodity to the combined dataframe.
+            combined = pd.concat([combined, df])
+
+    # Sanity check.
+    assert set([column for column in combined.columns if "value" in column.lower()]) == {
+        "Unit value $/t",
+        "Unit value 98$/t",
+    }
+    return combined
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap = paths.load_snapshot("historical_statistics_for_mineral_and_material_commodities.zip")
+
+    # Gather data (as ExcelFile objects) for all commodities.
+    supply_demand_data = read_data_for_all_commodities(snap=snap)
+
+    #
+    # Process data.
+    #
+    # Process and combine data for all commodities.
+    combined = combine_data_for_all_commodities(supply_demand_data=supply_demand_data)
+
+    # Create a table with metadata.
+    tb = pr.read_from_df(data=combined, metadata=snap.to_table_metadata(), origin=snap.metadata.origin, underscore=True)
+
+    # Columns contain a mix of numbers and strings (I suppose corresponding to footnotes).
+    # For now, save all data as strings.
+    tb = tb.astype({column: str for column in tb.columns})
```

Comment:
Hmmm, if I store the "year" column as a `str`, ETL (or, I guess, the repack library?) will pack it as an integer. But if I store it as `"string"`, then it's stored as a real string (and the following step fails). So, shall we adapt repack to also be smart about `"string"` columns? @Marigold (not urgent)
For now, I'll leave the code with `str`, to avoid the issues.

Reply:
Good point. `repack` indeed leaves `"string"` type alone and doesn't convert it to int / categorical. I think we should be consistent and repack it in the same way as we repack `str` / `object`